derive/src/utils.rs

```diff
@@ -458,12 +485,9 @@ fn check_variant_attribute(attr: &Attribute) -> syn::Result<()> {
 
 			Meta::NameValue(MetaNameValue {
 				path,
-				value: Expr::Lit(ExprLit { lit: Lit::Int(lit_int), .. }),
+				value: Expr::Lit(ExprLit { lit: Lit::Int(_), .. }),
 				..
-			}) if path.get_ident().map_or(false, |i| i == "index") => lit_int
-				.base10_parse::<u8>()
-				.map(|_| ())
-				.map_err(|_| syn::Error::new(lit_int.span(), "Index must be in 0..255")),
```

Comment:
In theory this check is still necessary, when we call `variant_index`:
https://github.com/paritytech/parity-scale-codec/blob/dacc95e2dc8eeb84b94175670b63cb7db1f09327/derive/src/utils.rs#L138-L140

somebody writing an index more than `2^64` will make the macro panic.

Though quite unlikely.